openmc/mgxs/mdgxs.py

```diff
@@ -0,0 +1,2619 @@
+            A boolean representing whether to eliminate the extra dimensions
+            of the multi-dimensional array to be returned. Defaults to True.
+
+        Returns
+        -------
+        numpy.ndarray
+            A NumPy array of the multi-group and multi-delayed-group cross
+            section indexed in the order each group, subdomain and nuclide is
+            listed in the parameters.
+
+        Raises
+        ------
+        ValueError
+            When this method is called before the multi-group cross section is
+            computed from tally data.
+
+        """
+
+        cv.check_value('value', value, ['mean', 'std_dev', 'rel_err'])
+        cv.check_value('xs_type', xs_type, ['macro', 'micro'])
+
+        # FIXME: Unable to get microscopic xs for mesh domain because the mesh
+        # cells do not know the nuclide densities in each mesh cell.
+        if self.domain_type == 'mesh' and xs_type == 'micro':
+            msg = 'Unable to get micro xs for mesh domain since the mesh ' \
+                  'cells do not know the nuclide densities in each mesh cell.'
+            raise ValueError(msg)
+
+        filters = []
+        filter_bins = []
+
+        # Construct a collection of the domain filter bins
+        if not isinstance(subdomains, str):
+            cv.check_iterable_type('subdomains', subdomains, Integral,
+                                   max_depth=3)
+            for subdomain in subdomains:
+                filters.append(_DOMAIN_TO_FILTER[self.domain_type])
+                filter_bins.append((subdomain,))
+
+        # Construct list of energy group bounds tuples for all requested groups
+        if not isinstance(groups, str):
+            cv.check_iterable_type('groups', groups, Integral)
+            for group in groups:
+                filters.append(openmc.EnergyoutFilter)
+                filter_bins.append(
+                    (self.energy_groups.get_group_bounds(group),))
+
+        # Construct list of delayed group tuples for all requested groups
+        if not isinstance(delayed_groups, str):
+            cv.check_type('delayed groups', delayed_groups, list, int)
+            for delayed_group in delayed_groups:
+                filters.append(openmc.DelayedGroupFilter)
+                filter_bins.append((delayed_group,))
+
+        # If chi delayed was computed for each nuclide in the domain
+        if self.by_nuclide:
+
+            # Get the sum as the fission source weighted average chi for all
+            # nuclides in the domain
+            if nuclides == 'sum' or nuclides == ['sum']:
+
+                # Retrieve the fission production tallies
+                delayed_nu_fission_in = self.tallies['delayed-nu-fission-in']
+                delayed_nu_fission_out = self.tallies['delayed-nu-fission-out']
+
+                # Sum out all nuclides
+                nuclides = self.get_nuclides()
+                delayed_nu_fission_in = delayed_nu_fission_in.summation\
```

Comment:
Nit: It's not strictly non-PEP8, but it's very uncommon to line break between the function name and the parenthesis of the function call.  Both for here and below.

There are two alternatives I would recommend:

Compress it to one line: 
`delayed_nu_fission_in.summation(nuclides=nuclides)`

Do a line break after the parenthesis, rather than before.  Example:

```
delayed_nu_fission_in.summation(
   nuclides=nuclides
)
```